ocean/src/clans.rs

```diff
@@ -1,39 +1,57 @@
+use std::collections::{HashMap, HashSet};
 
 #[derive(Debug)]
 pub struct ClanSystem {
     // TODO: add necessary fields
+    clans: HashMap<String, HashSet<String>>, 
 }
 
 impl ClanSystem {
     pub fn new() -> ClanSystem {
-        unimplemented!();
+        ClanSystem {clans: HashMap::new()}
     }
 
     /**
      * Returns a list of the names of the clan members for the given clan id.
      */
     pub fn get_clan_member_names(&self, clan_id: &str) -> Vec<String> {
-        unimplemented!();
+        //unimplemented!();
+        if let Some(members) = self.clans.get(clan_id) {
+            members.iter().cloned().collect()
+        } else {
+            Vec::new()
+        }
     }
 
     /**
      * Returns the number of clans currently in existence.
      */
     pub fn get_clan_count(&self) -> usize {
-        unimplemented!();
+        //unimplemented!();
+        self.clans.len()
     }
 
     /**
      * Returns the number of clan members for the given clan id.
      */
     pub fn get_clan_member_count(&self, clan_id: &str) -> usize {
-        unimplemented!();
+        //unimplemented!();
+        if let Some(members) = self.clans.get(clan_id) {
+            members.len()
+        } else {
+            0
+        }
     }
 
     /**
      * Returns the id of the clan with the most number of members, or None if such a clan does not exist.
      */
     pub fn get_largest_clan_id(&self) -> Option<String> {
-        unimplemented!();
+        let largest_clan = self.clans.iter().max_by_key(|(_, members)| members.len());
+        largest_clan.map(|(clan_id, _)| clan_id.clone())
+    }
+    pub fn add_member_to_clan(&mut self, clan_id: String, crab_name: String) {
```

Comment:
Introduce some spacing between each function, and consider including a brief description for the function.